src/client/client.cpp

```diff
@@ -137,7 +139,12 @@ namespace OpcUa
         }
         for (  UserTokenPolicy token : ed.UserIdentifyTokens)
         {
-          if (token.TokenType == UserIdentifyTokenType::ANONYMOUS )
+          if (token.TokenType == UserIdentifyTokenType::USERNAME && has_login)
```

Comment:
Let completely rely on TokenType and do not check login here?
Login should check OS for example.